### PR TITLE
Kernel Live Patching concept

### PR DESCRIPTION
**Pull request by dmpop**

### CODE REVIEW

File: xml/concept-klp.xml [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+<!-- This file can be edited downstream. -->
+<?xml-stylesheet href="urn:x-suse:xslt:profiling:docbook51-profile.xsl"
+ type="text/xml"
+ title="Profiling step"?>
+<!DOCTYPE article
+[
+  <!ENTITY % entities SYSTEM "generic-entities.ent">
+    %entities;
+]>
+<article xml:id="concept-klp" xml:lang="en"
```
Comment (tomschr) [inline xml/concept-klp.xml:12]: It seems, our styleguide prefers the longer version:

```suggestion
<article xml:id="concept-kernel-live-patching" xml:lang="en"
```

File: xml/concept-klp.xml [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  <!--example: how-it-works-ntp-time-synchronization-->
+  <title>How does &klp; work?</title>
+  <para>
+   Kernel live patches are delivered as packages with modified code that are
+   separate from the main kernel package. The live patches are cumulative, so
+   the latest patch contains all fixes from the previous ones for the kernel
+   package. Each kernel live package is tied to the exact kernel revision for
+   which it is issued. The live patch package version number increases with
+   every addition of fixes.
+  </para>
```
Comment (tomschr) [inline xml/concept-klp.xml:91]: Question: so a version `KERNEL_VERSION-2` is the second live patch package, right? 

Additionally, I had to read it twice to understand that there are two versions you are talking: kernel version plus live patch version. Is that right? (Maybe it was just my digestion which kicked in. :wink: )

File: xml/concept-klp.xml [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+   <para>
+    Live patches contain only critical fixes, and they do not replace regular
+    kernel updates that require a reboot. Consider live patches as temporary
+    measures that protect the kernel until a proper kernel update and a reboot
+    are performed.
+   </para>
+  </important>
+  <para>
+   The diagram below illustrates the overall relationship between live patches
+   and kernel updates. The list of CVEs and defect reports addressed by the
```
Comment (tomschr) [inline xml/concept-klp.xml:103]: maybe use the long name **and** the abbreviated term for the first time it's mentioned?

```suggestion
   and kernel updates. The list of <emphasis>Common Vulnerabilities and Exposures</emphasis> (CVEs) and defect reports addressed by the
```

File: xml/concept-klp.xml [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+   currently active live patch can be viewed using the <command>klp -v
+   patches</command> command.
+  </para>
+  <informalfigure>
+   <mediaobject>
+    <imageobject role="fo">
+     <imagedata fileref="klp.png" width="80%" format="PNG"/>
+    </imageobject>
+    <imageobject role="html">
+     <imagedata fileref="klp.png" width="100%" format="PNG"/>
```
Comment (tomschr) [inline xml/concept-klp.xml:113]: Remove the format attribute as this is a) unnecessary as the stylesheets derive the format from the filename extension, and b) will lead to validation errors in the future when validating with GeekoDoc v2:

```suggestion
     <imagedata fileref="klp.png" width="80%"/>
    </imageobject>
    <imageobject role="html">
     <imagedata fileref="klp.png" width="100%"/>
```

File: xml/concept-klp.xml [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+   With &klpa; activated, every kernel update comes with a live patch package.
+   This live patch does not contain any fixes and serves as a seed for future
+   live patches for the corresponding kernel. These empty seed patches are
+   called <literal>initial patches</literal>.
+  </para>
+ </section>
+ <section xml:id="benefits-klp">
+  <title>Benefits of &klp;</title>
+  <para>
+   &klpa; offers several benefits.
```
Comment (tomschr) [inline xml/concept-klp.xml:135]: Not entirely sure, but doesn't the styleguide recommends to introduce the items of a list with a colon?

```suggestion
   &klpa; offers several benefits:
```

File: xml/concept-klp.xml [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Performing Kernel Live Patching
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Troubleshooting Kernel Live Patching issues
```
Comment (tomschr) [inline xml/concept-klp.xml:182]: Are these titles of corresponding tasks? In that case, I'd recommend to use `<citetitle>`.

**Review by tomschr — COMMENTED** [review]
@dmpop This is a wonderful concept, well done. :+1:  I've have only minor suggestions.

File: xml/concept-klp.xml (hunk 2 above)
Comment (dmpop) [reply xml/concept-klp.xml:91]: There is a figure that (hopefully) clarifies things. Anyway, this description has been approved by the devs, and I'm reluctant to change it.

File: xml/concept-klp.xml (hunk 1 above)
Comment (sknorr) [reply xml/concept-klp.xml:12]: I think SLE-KLP is marketed as a "product" with an "official" acronym, so I guess the short version is probably fine.

File: xml/concept-klp.xml (hunk 5 above)
Comment (dmpop) [reply xml/concept-klp.xml:135]: Yes, _items_ not full sentences. So this is correct:

```
X offers the following advantages:

- speed
- cost saving
```

BUT

```
X offers the following advantages.

- Increase in speed.
- Significant reduction is costs.
```

Not the best example, but you get the idea.

**Pull request merged**